toolkit/tools/bldtracker/bldtracker.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Tool to track build time during image generation.
+
+package main
+
+import (
+	"encoding/csv"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	app          = kingpin.New("bldtracker", "A tool that helps track build time of different steps in the makefile.")
+	scriptName   = app.Flag("script-name", "The name of the current tool.").Required().String()
+	stepName     = app.Flag("step-name", "The name of the current step.").Required().String()
+	actionName   = app.Flag("action-name", "The name of the current sub-action.").Default("").String()
+	filePath     = app.Flag("file-path", "The folder that stores timestamp csvs.").Required().String()                          // currently must be absolute
+	mode         = app.Flag("mode", "The mode of this tool. Could be 'initialize' ('n') or 'record'('r').").Required().String() // should I set a default?
+	completePath string
+)
+
+func main() {
+	app.Version(exe.ToolkitVersion)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+	idx := strings.Index(*scriptName, ".")
+	var shortName string
+	if idx < 0 { // if scriptName does not have a suffix
+		shortName = *scriptName
+	} else { // if scriptName has a suffix
+		shortName = (*scriptName)[:idx]
+	}
```

Comment:
I think we could use the [`Cut` method](https://pkg.go.dev/strings#Cut) to achieve this in a more concise manner- something like the following:

```suggestion
	var shortName string
        shortName, _, _ = strings.Cut((*scriptName), ".")
```

Reply:
I tried to run bldtracker.go with this edit, and it gives me an error "undefined: strings.Cut". My guess is that maybe we are using an older version of the strings package. Daniel mentions that linux is able to deal with file names that has multiple "."s (ie. create_worker_chroot.sh.csv), so I think I will just not try to remove the suffix here for now.

Reply:
Ah, `strings.Cut` was introduced in 1.18- missed that when looking at the docs.